src/lightning/data/streaming/data_processor.py

```diff
@@ -784,6 +792,8 @@ def __init__(
             random_seed: The random seed to be set before shuffling the data.
             reorder_files: By default, reorders the files by file size to distribute work equally among all workers.
                 Set this to ``False`` if the order in which samples are processed should be preserved.
+            weights: Provide a list of weights associated to the inputs.
```

Comment:
can there be a way to auto-calculate the weights for even distribution?

Reply:
If the inputs contain files, yes. But this isn't always the case.

Reply:
My current use case requires can't be auto-calculated.